js/src/common/utils/Drawer.js

```diff
@@ -1,20 +1,41 @@
+import { createFocusTrap } from './focusTrap';
+import { throttle } from './throttleDebounce';
+
 /**
  * The `Drawer` class controls the page's drawer. The drawer is the area the
  * slides out from the left on mobile devices; it contains the header and the
  * footer.
  */
 export default class Drawer {
+  /**
+   * @type {import('./focusTrap').FocusTrap}
+   */
+  focusTrap;
+
   constructor() {
     // Set up an event handler so that whenever the content area is tapped,
     // the drawer will close.
-    $('#content').click((e) => {
+    $('#content').on('click', (e) => {
       if (this.isOpen()) {
         e.preventDefault();
         this.hide();
       }
     });
+
+    this.focusTrap = createFocusTrap('#drawer', { allowOutsideClick: true });
+  }
+
+  resizeHandler(e) {
+    if (this.isOpen()) {
+      if (document.documentElement.style.getPropertyValue('--flarum-screen') !== 'phone') {
```

Comment:
Can we use the `app.screen` encapsulation?

Reply:
Ooh, I didn't realise that existed! Thanks for pointing it out!